platforms/core-execution/persistent-cache/src/main/java/org/gradle/cache/internal/DefaultPersistentDirectoryStore.java

```diff
@@ -202,17 +204,24 @@ private boolean requiresCleanup() {
         public void cleanup() {
             if (cacheCleanupStrategy != null && requiresCleanup()) {
                 String description = "Cleaning " + getDisplayName();
-                ProgressLogger progressLogger = progressLoggerFactory.newOperation(CacheCleanupExecutor.class).start(description, description);
-                Timer timer = Time.startTimer();
-                try {
-                    cacheCleanupStrategy.getCleanupAction().clean(DefaultPersistentDirectoryStore.this, new DefaultCleanupProgressMonitor(progressLogger));
-                    GFileUtils.touch(gcFile);
-                } finally {
-                    LOGGER.info("{} cleaned up in {}.", DefaultPersistentDirectoryStore.this, timer.getElapsed());
-                    progressLogger.completed();
-                }
+                buildOperationRunner.run(new RunnableBuildOperation() {
+                    @Override
+                    public void run(BuildOperationContext context) {
+                        Timer timer = Time.startTimer();
+                        try {
+                            cacheCleanupStrategy.getCleanupAction().clean(DefaultPersistentDirectoryStore.this, new DefaultCleanupProgressMonitor(context));
+                            GFileUtils.touch(gcFile);
+                        } finally {
+                            LOGGER.info("{} cleaned up in {}.", DefaultPersistentDirectoryStore.this, timer.getElapsed());
+                        }
+                    }
+
+                    @Override
+                    public BuildOperationDescriptor.Builder description() {
+                        return BuildOperationDescriptor.displayName(displayName);
+                    }
+                });
```

Comment:
💭 This is the main change in the PR. The rest is mostly to update services and tests to keep functioning.